src/main/java/com/minecolonies/network/messages/GuardTaskMessage.java

```diff
@@ -0,0 +1,106 @@
+package com.minecolonies.network.messages;
+
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.ColonyManager;
+import com.minecolonies.colony.buildings.BuildingGuardTower;
+import com.minecolonies.colony.permissions.Permissions;
+import com.minecolonies.util.BlockPosUtil;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Message to change the behavior of the guards..
+ */
+public class GuardTaskMessage extends AbstractMessage<GuardTaskMessage, IMessage>
+{
+    private int      colonyId;
+    private BlockPos buildingId;
+    private int job;
+    private boolean  assignmentMode;
+    private boolean  patrollingMode;
+    private boolean retrieval;
+    private int task;
+
+    /**
+     * Empty standard constructor.
+     */
+    public GuardTaskMessage()
+    {
+        super();
+    }
+
+    /**
+     * Creates object for the assignmentMode message.
+     *
+     * @param building       View of the building to read data from.
+     * @param assignmentMode assignmentMode of the particular farmer.
+     */
+    public GuardTaskMessage(@NotNull BuildingGuardTower.View building, int job, boolean assignmentMode, boolean patrollingMode, boolean retrieval, int task )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document the parameter(s): job, patrollingMode, retrieval, task [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)